src/features/inlay_hints.zig

```diff
@@ -254,22 +268,56 @@ fn writeVariableDeclHint(builder: *Builder, decl_node: Ast.Node.Index) !void {
         &type_str,
         &reference_collector,
     );
-    if (type_str.len == 0) return;
+    if (type_str.len == 0) return null;
 
-    // TODO: Remove once long type hints can be reduced, i.e. `struct { .. }`
```

Comment:
Was it intentional to remove this TODO comment?

Reply:
Wouldn't this be as simple as,
if hint startsWith "struct " => hint = "struct {...}"
if hint startsWith "fn " => hint = "fn"
enum, union(enum), ..

Reply:
It may be reasonable to show small containers like `struct { foo: u32 }` or `struct { []const u8 }` but except that, I think your right.

Reply:
I added the comment back in, it would be a topic for a separate PR though.